compass/landice/tests/thwaites/uq_ensemble/__init__.py

```diff
@@ -0,0 +1,52 @@
+from compass.validate import compare_variables
+from compass.testcase import TestCase
+from compass.landice.tests.thwaites.uq_ensemble.ensemble_member \
+        import EnsembleMember
+from compass.landice.tests.thwaites.uq_ensemble.ensemble_manager \
+        import EnsembleManager
+
+
+class UQEnsemble(TestCase):
+    """
+    A test case for performing two MALI runs of the Thwaites setup,
+    with two different core counts.  The test case verifies that the
+    results of the two runs are identical.
+    """
+
+    def __init__(self, test_group):
+        """
+        Create the test case
+
+        Parameters
+        ----------
+        test_group : compass.landice.tests.thwaites.Thwaites
+            The test group that this test case belongs to
+
+        """
+        name = 'thwaites_uq'
+        super().__init__(test_group=test_group, name=name)
+
+        # We need the list of runs to set up steps.
+        # Steps need to be defined in TestCase.__init__
+        # But we want to keep operations to a minimum
+
+        self.nRuns = 20 # number of runs in ensemble
```

Comment:
@xylar , in our discussion, I believe you suggested the number of runs could come from the config file.  But my understanding is the config is not parsed until the `configure`/`setup` step.  Can you explain or point me to an example of how best to get this value from the config here in the constructor?

Reply:
You can (and I think should in this circumstance) pass a user config file with `-f` as part of `compass setup`.  That would be the config file that sets a config option.  This config option could default to 20 (or 1 or whatever makes sense as a default).  Ideally, users would also use the config file to set other properties of the ensemble *before* setting it up.  This would be qualitatively similar to the config file required for the `ismip6_forcing` test cases.

Reply:
@xylar , I agree with the idea of having a cfg file where some user adjustments are expected prior to setting up the test case for this. I took a look at the ismip6_forcing test group again, but I can't find an example of where the config is parsed in the constructor.  In order to init the steps for the, say, 20 runs, we need to know the number of runs in the test case constructor, but standard practice is that the config object is not parsed and available until `setup`.  From what I can tell, the ismip6_forcing group follows that standard practice, but I might be missing something.

(BTW, you may have noticed this comment thread only appears in the conversation tab of the PR, because I have subsequently changed this line of code for other reasons.)

Reply:
The config file is never parsed in the constructor.  That's not possible because that happens even for compass list.  It must happen in the `setup()` method.  You have to move this stuff from the constructor to `configure()`.

Reply:
Take a look at the cosine bell test case for an example of this:
https://github.com/MPAS-Dev/compass/blob/main/compass/ocean/tests/global_convergence/cosine_bell/__init__.py

We start with a default set of steps based on the default value of the config option `resolutions`, but then we throw away those steps at `configure()` and make a new set based on a modified set of resolutions (if necessary).

Reply:
In your case, I wouldn't bother with the stuff in `__init__()` because you should be fine with there being zero steps in the test case by default (e.g. in `compss list --verbose`).  You can just add the steps within `configure()`.

Reply:
Ah, I see.  I was so focused on trying to get the steps finalized in the constructor.  I know you keep referring me to the cosine bell test, and now I understand that aspect of it.  I had understood that `steps_to_run` could be changed at setup time, but I hadn't made the connection that the actual set of steps that are part of the test case could be entirely adjusted during setup.  I'll take a closer look and implement that when I have time to get back to this.  Thanks for the explanation.

Reply:
Changed in 974353cb81252add490a5ede2c939eb878cf7567